packages/client/tests/functional/concurrent-transaction-unique/tests.ts

```diff
@@ -0,0 +1,45 @@
+// @ts-ignore
+import { PrismaClient } from '@prisma/client'
+import crypto from 'crypto'
+
+import testMatrix from './_matrix'
+
+declare let prisma: PrismaClient
+
+const data = [
+  { name: 'name', id: crypto.randomBytes(12).toString('hex') },
+  { name: 'name', id: crypto.randomBytes(12).toString('hex') },
+  { name: 'name', id: crypto.randomBytes(12).toString('hex') },
+]
+
+/**
+ * Reproduction for issue #9678
+ */
+testMatrix.setupTestSuite(({ provider }) => {
+  testIf(provider !== 'sqlite')('serialized deleteMany/createMany', async () => {
+    const fn = async () => {
+      await prisma.$transaction([
+        prisma.resource.deleteMany({ where: { name: 'name' } }),
+        prisma.resource.createMany({ data }),
+      ])
+    }
+
+    for (let i = 0; i < 50; i++) {
+      await fn()
+    }
+  })
+
+  testIf(provider !== 'sqlite' && provider !== 'mysql' && provider !== 'sqlserver').failing(
```

Comment:
Why exclude SQlite, MySQL and SQL Server?

Reply:
Reply via Slack: Because there it does not fail as described (and tested here).

Reply:
I assume we do not have a nice way to test that it fails on Mongo, POstgres and Cockroach but _succeeds_ for the others?

Reply:
No, I couldn't come up with something better